spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala

```diff
@@ -70,6 +70,12 @@ class DeltaCatalog extends DelegatingCatalogExtension
 
   val spark = SparkSession.active
 
+  private lazy val isUnityCatalog: Boolean = {
+    val delegateField = classOf[DelegatingCatalogExtension].getDeclaredField("delegate")
+    delegateField.setAccessible(true)
+    delegateField.get(this).getClass.getCanonicalName.startsWith("io.unitycatalog")
```

Comment:
```suggestion
    delegateField.get(this).getClass.getCanonicalName.startsWith("io.unitycatalog.")
```
... to at least slightly reduce the chances of a false positive?